### DISCUSSION TITLE
How to assign a variable to a canvas property?

### DISCUSSION THREAD

**Discussion by ghost** [fabricjs/fabric.js#7783] [Q&A] [answered]
I have three buttons to change pencil thickness for free drawing mode. The onclick listener for these three buttons is setPencilThickness(). At program start, the values for smallThickness, mediumThickness, largeThickness are 4, 8, 12 respectively. Since, I use this function internally as well to programmatically change the pencil thickness on selecting free drawing mode, I have an optional thickness paramter, which can be one of small-thickness, medium-thickness, large-thickness. This function works perfectly fine.
However, the issue came when I implemented a settings function, where I have three HTML input fields, in which I can type new value for smallThickness, mediumThickness, largeThickness. The function that gets executed when the user finishes typing these new values is applySettings(). The code for both of these functions is given below. For the settings to take effect, I have to call the setPencilThickness() function again (which I am currently just manually doing by clicking on the buttons -  I was planning to write proper code for it). However, the output is not as expected! Initially, the freeDrawingBrush width DOES CHANGE CORRECTLY but with lots of weird things happening, such as drawn paths shifting positions, sometimes get deleted on their own etc etc. More-ever a lot of other things break as well. For example, when I try to rotate the rectangles on the canvas, their border and controls get mis-aligned. Can someone explain how to implement the settings function correctly. I will post my project on github.io in the next comment and give instructions on how to replicate the problem. Thanks a lot in advance :)

`function setPencilThickness(e, thickness = undefined) {
    const thicknessButtons = ["small-thickness", "medium-thickness", "large-thickness"];

    thicknessButtons.forEach((item) => {
        const t = document.getElementById(item);
        t.classList.remove("active");
    });

    if (thickness === undefined) thickness = e.currentTarget.id;
    const t = document.getElementById(thickness);
    t.classList.add("active");

    switch (thickness) {
        case "small-thickness":
            v = smallThickness;
            break;
        case "medium-thickness":
            v = mediumThickness;
            break;
        case "large-thickness":
            v = largeThickness;
            break;
        default:
            v = smallThickness;
    }

    pencilThickness = thickness;
    canvas.freeDrawingBrush.width = v;
}`

`function applySettings() {
    snap = document.getElementById("grid-snap").value;
    rotateAngle = document.getElementById("rotate-snap").value;
    pasteOffset = document.getElementById("paste-offset").value;
    smallThickness = document.getElementById("small-thickness-input").value;
    mediumThickness = document.getElementById("medium-thickness-input").value;
    largeThickness = document.getElementById("large-thickness-input").value;
}`

**Comment by ghost** [comment, marked as answer]
> 

The link to my app: https://akaabdullahmateen.github.io/
The source code link: https://github.com/akaAbdullahMateen/akaAbdullahMateen.github.io
Steps to reproduce the error:
1) Select pencil tool from the bottom toolbox (second button from left) or press 2.
2) A menu will appear: the first three buttons from the left side are the concerned buttons for settings small thickness, medium thickness, large thickness. Click any one of them, and optionally, select a color from the rest of the menu and start drawing on canvas. Verify everything works as intended.
3) Now click on the settings button on the top-right corner (gear icon). A sidebar will open.
4) In the Pencil category, change the value for small thickness, (for example, change 4 to 2) and click on apply button in the bottom right corner of the sidebar.
5) Close the sidebar by either clicking on the cross button in the top-right of sidebar. Or the sidebar button next to the bottom right corner of the sidebar.
6) Select pencil tool again (in case you selected another tool for testing for example, pointer, eraser etc).
7) Click on the small thickness button (the left-most button in the menu that appears above the toolbox).
8) And start drawing on canvas again :)
9) Notice some problems? 
10) Right click on any one of the two rectangles already present.
11) Select "rotate clockwise" or press CTRL+K (I added the mac command icon in shortcut instead of CTRL because it looked cool XD, it is CTRL key NOT COMMAND key).
12) The rectangle should rotate.
13) Keep pressing CTRL+K about 5 or more times.
14) Notice the border and controls of the rectangle mis-aligning?

The app is incomplete, so a lot of buttons dont work :)

**Reply by melchiar** [reply]
Are you sure your values are the right data type?

Numbers grabbed from an input field are usually strings, so you'd need to do `parseInt()` on them before assigning them.

If it's not that, can you share a fiddle or codepen showing the problem?

**Reply by ghost** [reply]
I can't create the codepen or fiddle, sorry. I did share the link to github repository with everything in it.
Also, I will try the parseInt() thing. Give me a minute.

**Reply by ghost** [reply]
OMGGGGGGGGGGGGGGGGG
Are you a genius or something ? 
It did fix the problem xddddddddddddddddddddddddddddd

**Reply by ghost** [reply]
Thanks a lot for the help :)
I m really noob, omg.

**Reply by ghost** [reply]
Can you type your comment in a answer box so I can mark it as answer?

**Reply by melchiar** [reply]
sure

**Comment by melchiar** [comment]
> Are you sure your values are the right data type?
> 
> Numbers grabbed from an input field are usually strings, so you'd need to do `parseInt()` on them before assigning them.
> 
> If it's not that, can you share a fiddle or codepen showing the problem?
